ServerCore/Pages/Puzzles/Index.cshtml.cs

```diff
@@ -19,11 +19,21 @@ public IndexModel(ServerCore.Models.PuzzleServerContext context)
             _context = context;
         }
 
-        public IList<Puzzle> Puzzle { get;set; }
+        public IList<Puzzle> Puzzle { get; set; }
 
-        public async Task OnGetAsync()
+        public int? EventId { get; set; }
+
+        public async Task OnGetAsync(int? eventid)
         {
-            Puzzle = await _context.Puzzle.ToListAsync();
+            if (eventid != null)
```

Comment:
Please file an issue tracking removing the null tolerance since we don't want people accidentally wandering into all puzzles in all events